src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Transition.kt

```diff
@@ -73,16 +75,21 @@ data class Conditions(
     val docCount: Long?,
     val size: String?,
     val cron: CronSchedule?
-) : ToXContent {
+) : ToXContentObject {
 
     init {
-        var conditionsSet = 0
-        if (indexAge != null) conditionsSet++
-        if (docCount != null) conditionsSet++
-        if (size != null) conditionsSet++
-        if (cron != null) conditionsSet++
+        val conditionsList = listOf(indexAge, docCount, size, cron)
+        require(conditionsList.filterNotNull().size < 2) { "Cannot provide more than one Transition condition" }
+
+        // Validate indexAge condition
+        try {
+            TimeValue.parseTimeValue(indexAge, "")
+        } catch (e: IllegalArgumentException) {
+            throw IllegalArgumentException("Must have a valid index age Transition condition")
+        }
 
-        require(conditionsSet < 2) { "Cannot provide more than one Transition condition" }
+        // Validate size condition
+        requireNotNull(size?.toIntOrNull()) { "Must have a valid size Transition condition" }
```

Comment:
This will always throw an exception when the person does not use size, wrap in if (size != null)

Let's add some tests to validate these work correctly.

Reply:
Changed and added tests for validation.